### PR TITLE
[#3911] Add `SaveActivity` data and sheet

### PR DESCRIPTION
**Pull request by arbron**
Adds a new activity type for making saves and applying damage. This involves refactoring out some code from the attack & summons sheets into `ActivitySheet` for some shared behavior.

Save Sheet V1 - Effect

Also reworks the damage parts list and the applied effects into a `.separated-list` like on the summons sheet.

Note: Does not contain any functionality for rolling saves or damage.

### Todo
- [ ] Add method for forwarding ability saves to actor
- [ ] Add `rollDamage` (shared with other activities)
- [ ] Properly communicate "Half Damage", etc. on chat cards

### CODE REVIEW

File: module/data/activity/base-activity.mjs [hunk 1]
```diff
@@ -129,9 +129,7 @@ export default class BaseActivityData extends foundry.abstract.DataModel {
         units: new StringField({ initial: "inst" }),
         special: new StringField()
       }),
-      effects: new ArrayField(new SchemaField({
-        id: new DocumentIdField()
-      })),
+      effects: new ArrayField(new SchemaField(this.defineEffectSchema())),
```
Comment (Fyorl) [inline module/data/activity/base-activity.mjs:132]: Maybe better as its own field that extends `SchemaField`:

```js
class ActivityEffectData extends SchemaField {
  constructor(fields={}, options={}) {
    super({
      id: new DocumentIdField(),
      ...fields
    }, options);
  }
}
```

File: module/data/activity/save-data.mjs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+/**
+ * @typedef {EffectApplicationData} SaveEffectApplicationData
+ * @property {string} onSave  Should this effect still be applied on a successful save?
+ */
+
+/**
+ * Data model for an save activity.
+ *
+ * @property {string} ability                       Ability used to make the attack and determine damage.
```
Comment (Fyorl) [inline module/data/activity/save-data.mjs:15]:
```suggestion
 * @property {string} ability                       Make the saving throw with this ability.
```

File: templates/activity/parts/activity-effects.hbs [hunk 3]
```diff
@@ -3,18 +3,43 @@
...
+    {{{ dnd5e-linkForUuid effect.uuid }}}
+    <div class="list-controls flexrow">
+        <button type="button" class="unbutton" data-action="deleteEffect"
+                data-tooltip="DND5E.EFFECT.Action.Delete"
+                aria-label="{{ localize 'DND5E.EFFECT.Action.Delete' }}">
+            <i class="fa-solid fa-trash fa-fw" inert></i>
+        </button>
+    </div>
+</div>
+<input type="hidden" name="{{ prefix }}id" value="{{ data.id }}">
```
Comment (Fyorl) [inline templates/activity/parts/activity-effects.hbs:30]: Missing a dot, but maybe this should be `_id` as well? Not sure.
```suggestion
<input type="hidden" name="{{ prefix }}.id" value="{{ data.id }}">
```

File: templates/activity/parts/save-details.hbs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<fieldset>
+    <legend>{{ localize "DND5E.SAVE.FIELDS.save.label" }}</legend>
+    {{ formField fields.ability value=source.ability options=abilityOptions }}
+    {{#with fields.save.fields.dc.fields as |fields|}}
+    {{ formField fields.calculation value=../source.save.dc.calculation options=../calculationOptions }}
+    {{#if (eq ../source.save.dc.calculation "custom")}}
+    {{ formField fields.formula value=../source.save.dc.formula }}
+    {{else}}
+    {{ formField fields.formula name="" value=(localize "DND5E.SAVE.FIELDS.save.dc.DefaultFormula") disabled=true }}
+    <input type="hidden" name="save.dc.formula" value="{{ ../source.save.dv.formula }}">
```
Comment (Fyorl) [inline templates/activity/parts/save-details.hbs:10]:
```suggestion
    <input type="hidden" name="save.dc.formula" value="{{ ../source.save.dc.formula }}">
```

**Review by Fyorl — CHANGES_REQUESTED** [review]
Looks pretty good, thank you

File: templates/activity/parts/activity-effects.hbs (hunk 3 above)
Comment (arbron) [reply templates/activity/parts/activity-effects.hbs:30]: The prefix contains the dot `effects.${index}.`.

**Review by Fyorl — APPROVED** [review]

**Pull request merged**